code/modules/reagents/reagent_dispenser.dm

```diff
@@ -390,4 +390,105 @@
+	reagents.reaction(usr.loc)
+	src.reagents.clear_reagents()
+	H.visible_message("<span class='warning'>[usr] kicks \the [src]!</span>", "<span class='notice'>You kick \the [src].</span>")
+	for(var/atom/movable/AM in src)
+		AM.forceMove(loc)
+
+/obj/structure/reagent_dispensers/cauldron/barrel/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(iswrench(W))
+		return
+	if(istype(W,/obj/item/weapon/grab))
+		var/obj/item/weapon/grab/G = W
+		var/mob/living/target = G.affecting
+		user.visible_message("<span class='danger'>[user] begins to drag [target] into the barrel!</span>")
+		if(do_after_many(user,list(target,src),10)) //Twice the normal time
+			enter_barrel(target)
+	else
+		..()
+	return ..()
+
+/obj/structure/reagent_dispensers/cauldron/barrel/attackby(obj/item/W, mob/user, params)
+
+/obj/structure/reagent_dispensers/cauldron/barrel/proc/enter_barrel(mob/user)
+	user.forceMove(src)
+	update_icon()
+	user.reset_view()
+	to_chat(user,"<span class='notice'>You enter \the [src].</span>")
+
+/obj/structure/reagent_dispensers/cauldron/barrel/MouseDropTo(atom/movable/O, mob/user)
+	if(O.loc == user || !isturf(O.loc) || !isturf(user.loc) || !user.Adjacent(O)) //no you can't pull things out of your ass
+		return
+	if(user.incapacitated() || user.lying) //are you cuffed, dying, lying, stunned or other
+		return
+	if(!Adjacent(user) || !user.Adjacent(src)) // is the mob too far away from you, or are you too far away from the source
+		return
+	if(O.locked_to)
+		return
+	else if(O.anchored)
+		return
+	if(issilicon(O)) //robutts dont fit
+		return
+	if(!ishigherbeing(user) && !isrobot(user)) //No ghosts or mice putting people into the barrel
+		return
+	var/mob/living/target = O
+	if(!istype(target))
+		return
+	for(var/mob/living/carbon/slime/M in range(1,target))
+		if(M.Victim == target)
+			to_chat(user, "[target.name] will not fit into \the [src] because they have a slime latched onto their head.")
+			return
+
+	if(target == user)
+		to_chat(user,"<span class='notice'>You begin to climb into the barrel.</span>")
+		if(do_after(target,src,10))
+			enter_barrel(target)
+	else
+		user.visible_message("<span class='danger'>[user] begins to drag [target] into the barrel!</span>")
+		if(do_after_many(user,list(target,src),10)) //Twice the normal time
+			enter_barrel(target)
+
+/obj/structure/reagent_dispensers/cauldron/barrel/container_resist(mob/user)
+	if (exiting.Remove(user)) 
+		to_chat(user,"<span class='warning'>You stop climbing free of \the [src].</span>")
+		return
+	visible_message("<span class='warning'>[user] begins to climb free of the \the [src]!</span>")
+	exiting += user
+	spawn(3 SECONDS)
+		if(loc && exiting.Remove(user))
+			user.forceMove(loc)
+			update_icon()
+			to_chat(user,"<span class='notice'>You climb free of the barrel.</span>")
+
+/obj/structure/reagent_dispensers/cauldron/barrel/Destroy()
+	..()
```

Comment:
You need to call this after you forceMove the contents otherwise they will get deleted